src/components/App/index.tsx

```diff
@@ -252,19 +254,39 @@ export default class App extends Component<Props, State> {
     const image = images[index];
 
     let file;
-    try {
-      // Special case for identity
-      if (image.encoderState.type === identity.type) {
-        file = source.file;
-      } else {
-        if (!skipPreprocessing || !image.preprocessed) {
-          image.preprocessed = await preprocessImage(source, image.preprocessorState);
+    let preprocessed;
+    let bmp;
+    const cacheResult = this.encodeCache.match(source, image.preprocessorState, image.encoderState);
+
+    if (cacheResult) {
+      ({ file, preprocessed, bmp } = cacheResult);
+    } else {
+      try {
+        // Special case for identity
+        if (image.encoderState.type === identity.type) {
+          file = source.file;
+          bmp = source.bmp;
```

Comment:
I snuck a little fix in here. Previously we were passing `file` to the decoder in the identity case, meaning we were decoded something we already had decoded.